File: pset1/cash.c
//first asks the user how much change is owed and then prints the minimum number of coins with which that change can be made.
// 0.25, 0.10, 0.5, 0.1

#include <cs50.h>
#include <stdio.h>
#include <math.h>

float get_positive_float(void);

int main(void)
{
    float i = get_positive_float();
    int cents = round(i * 100);
    int coins = 0;
    while (cents > 0)
    {
        if (cents >= 25)
        {
            coins++;
            cents = cents - 25;
        }
        else if (cents >= 10)
        {
            coins++;
            cents = cents - 10;
        }
        else if (cents >= 5)
        {
            coins++;
            cents = cents - 5;
        }
        else
        {
            coins++;
            cents = cents - 1;
        }
    }
    printf("%i\n", coins);
}

float get_positive_float(void)
{
    float n;
    do
    {
        n = get_float("Change owed:  ");
    }
    while (n < 0);
    return n;
}

File: pset6/cash.py
from cs50 import get_float
from decimal import Decimal


def main():
    change = Decimal(get_positive_float())
    coins = 0
    cents = round(change * 100, 2)
    while cents > 0:
        if cents >= 25:
            coins += 1
            cents = cents - 25
        elif cents >= 10:
            coins += 1
            cents = cents - 10
        elif cents >= 5:
            coins += 1
            cents = cents - 5
        else:
            coins += 1
            cents = cents - 1
    print(coins)


def get_positive_float():
    while True:
        n = get_float("Change owed: ")
        if n > 0:
            break
    return n


main()
